content/advent-2016/cloudforest.md

```diff
@@ -0,0 +1,254 @@
+- `progress`: report the tree number and the running error. With `-progress`, it is possible to follow the construction of the forest. The first trees usually have high errors but the model quickly adjusts itself towards smaller and smaller errors as more trees are built. `-progress` is a good companion to `-nTrees`, it can be used to find out empirically how many trees are a good fit for your model. If the forest construction converges early, there is little benefit to increasing the number of trees.
+- `mTry`: the number of candidate features to be selected with each tree. Not every tree in a forest is build the same. Each classification tree is built with a subset of the features. `-mTry` determines how many features are available for selection during the construction of each tree. Lower values yield trees with less correlation and higher values allow more opportunities for important features to be selected. If not specified, `-mTry` is the square root of the number of features. This value is common across many implementations of Random Forests, and it's one of the parameters that has the highest impact on the accuracy of the final model. You may want to experiment with bigger values for `-mTry`. We've had success with much higher ratios, such as .5 (50% of the features available for each tree), but your mileage may vary.
+- `blacklist`: a list of features that shouldn't be used by any of the trees. Sometimes, parts of the data on the .afm files aren't really features, only some kind of metadata that is not relevant for classification. This option excludes these features without requiring you to edit the .afm file used on `growforest`.
+- `test`: a .afm file with more data used for testing. The data in this file is not used to generate the classification forest. After the model is complete, it is tested on this data and the accuracy results are reported in the end of `growforest`'s execution. This option is the equivalent of running `applyforest` right after finishing `growforest`.
+- `maxDepth`: depending on your data, some trees may have a hard time converging. `-maxDepth` determines how tall a tree can grow before the algorithm cuts it. Without `-maxDepth`, `-growforest` can run fast at first, completing the construction of several trees and then slow down in the end when some of the feature sets have trouble finding a good way to split the cases.
+
+# Importing and using CloudForest directly from other Go programs
+
+CloudForest was built as a Go package. We've seen only command line utilities so far, but these command line utilities are just `main` packages that import the base package and use its functions. Any other Go program can do the same.
+
+For example, loading an AFM file into memory can be done like this:
+
+
+```
```

Comment:
Great! I've done it for all the blocks. It helped me catch a mistake in the first block, too.